### PR TITLE
Annotation-work-/File-Cleaning-part-7

### PR DESCRIPTION
**Pull request by MrRowey**
Files from the `/lua/*` Directory
- EffectUtilities.lua
- EffectUtilitiesAeon.lua
- dataInit.lua
- defaultantiprojectile.lua
- defaultcollisionbeams.lua
- defaultexplosions.lua
- defaultunits.lua
- formations.lua
- game.lua
- kirvesbeams.lua

### CODE REVIEW

File: lua/EffectUtilities.lua [hunk 1]
```diff
@@ -229,7 +229,7 @@ end
 --- You can use `SpawnBuildBotsOpti` instead
 ---@param builder Unit
 ---@param unitBeingBuilt Unit
----@param buildEffectsBag any unused
+---@param buildEffectsBag any[] unused
```
Comment (Garanas) [inline lua/EffectUtilities.lua:232]: TrashBag

File: lua/defaultantiprojectile.lua [hunk 2]
```diff
@@ -12,6 +12,9 @@ local GetRandomFloat = import('/lua/utilities.lua').GetRandomFloat
 
 ---@class Flare : Entity
 Flare = Class(Entity){
+
+    ---@param self Flare
+    ---@param spec any
```
Comment (Garanas) [inline lua/defaultantiprojectile.lua:17]: You can decipher the spec based on what happens in `OnCreate` below, best to add in a separate class `FlareSpec`

File: lua/defaultantiprojectile.lua [hunk 3]
```diff
@@ -23,8 +26,11 @@ Flare = Class(Entity){
         self.RedirectCat = spec.Category or 'MISSILE'
     end,
 
-    -- We only divert projectiles. The flare-projectile itself will be responsible for
-    -- accepting the collision and causing the hostile projectile to impact.
+    --- We only divert projectiles. The flare-projectile itself will be responsible for
+    --- accepting the collision and causing the hostile projectile to impact.
+    ---@param self Flare
+    ---@param other any
```
Comment (Garanas) [inline lua/defaultantiprojectile.lua:32]: a Projectile

File: lua/defaultantiprojectile.lua [hunk 4]
```diff
@@ -35,6 +41,9 @@ Flare = Class(Entity){
 
 ---@class DepthCharge : Entity
 DepthCharge = Class(Entity) {
+
+    ---@param self DepthCharge
+    ---@param spec any
```
Comment (Garanas) [inline lua/defaultantiprojectile.lua:46]: You can decipher the spec based on what happens in `OnCreate` below, best to add a separate class `DepthChargeSpec`

File: lua/defaultantiprojectile.lua [hunk 5]
```diff
@@ -44,8 +53,11 @@ DepthCharge = Class(Entity) {
         self:AttachTo(spec.Owner, -1)
     end,
 
-    -- We only divert projectiles. The flare-projectile itself will be responsible for
-    -- accepting the collision and causing the hostile projectile to impact.
+    --- We only divert projectiles. The flare-projectile itself will be responsible for
+    --- accepting the collision and causing the hostile projectile to impact.
+    ---@param self DepthCharge
+    ---@param other any
```
Comment (Garanas) [inline lua/defaultantiprojectile.lua:59]: Projectile

File: lua/defaultantiprojectile.lua [hunk 6]
```diff
@@ -59,6 +71,8 @@ MissileRedirect = Class(Entity) {
         RedirectBeams = { '/effects/emitters/particle_cannon_beam_02_emit.bp' },
         EndPointEffects = {'/effects/emitters/particle_cannon_end_01_emit.bp' },
 
+        ---@param self MissileRedirect
+        ---@param spec any
```
Comment (Garanas) [inline lua/defaultantiprojectile.lua:75]: You can decipher the spec, see previous comments

File: lua/defaultcollisionbeams.lua [hunk 7]
```diff
@@ -92,10 +93,13 @@ QuantumBeamGeneratorCollisionBeam = Class(SCCollisionBeam) { -- used by CZAR
...
         '/effects/emitters/quantum_generator_04_emit.bp',
     },
-    
+
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.5,
 
+    ---@param self QuantumBeamGeneratorCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Projectile
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:102]: Unit | Projectile | Prop | nil

File: lua/defaultcollisionbeams.lua [hunk 8]
```diff
@@ -161,7 +167,8 @@ DisruptorBeamCollisionBeam = Class(SCCollisionBeam) {
 }
 
 ---@class MicrowaveLaserCollisionBeam01 : SCCollisionBeam
-MicrowaveLaserCollisionBeam01 = Class(SCCollisionBeam) { -- used by ML & cyb ACU
+MicrowaveLaserCollisionBeam01 = Class(SCCollisionBeam) {
+    -- used by ML & cyb ACU
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:171]: Best to add these to the class description, or they are lost essentially

File: lua/defaultcollisionbeams.lua [hunk 9]
```diff
@@ -171,6 +178,9 @@ MicrowaveLaserCollisionBeam01 = Class(SCCollisionBeam) { -- used by ML & cyb ACU
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.25,
 
+    ---@param self MicrowaveLaserCollisionBeam01
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:183]: `Unit | Projectile | Prop | nil`

File: lua/defaultcollisionbeams.lua [hunk 10]
```diff
@@ -237,6 +249,9 @@ PhasonLaserCollisionBeam = Class(SCCollisionBeam) { -- used by GC
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.25,
 
+    ---@param self PhasonLaserCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:254]: `Unit | Projectile | Prop | nil`

File: lua/defaultcollisionbeams.lua [hunk 11]
```diff
@@ -298,7 +313,8 @@ TractorClawCollisionBeam = Class(CollisionBeam) {
 --   QUANTUM BEAM GENERATOR COLLISION BEAM
 ------------------------------------
 ---@class ExperimentalPhasonLaserCollisionBeam : SCCollisionBeam
-ExperimentalPhasonLaserCollisionBeam = Class(SCCollisionBeam) { -- unknown unit (big size though)
+ExperimentalPhasonLaserCollisionBeam = Class(SCCollisionBeam) { 
+    -- unknown unit (big size though)
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:317]: See previous comments about this, best to add to class description

File: lua/defaultcollisionbeams.lua [hunk 12]
```diff
@@ -308,6 +324,9 @@ ExperimentalPhasonLaserCollisionBeam = Class(SCCollisionBeam) { -- unknown unit
     SplatTexture = 'scorch_004_albedo',
     ScorchSplatDropTime = 0.1,
 
+    ---@param self ExperimentalPhasonLaserCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:329]: Unit | Projectile | Prop | nil

File: lua/defaultcollisionbeams.lua [hunk 13]
```diff
@@ -378,6 +399,9 @@ UnstablePhasonLaserCollisionBeam = Class(SCCollisionBeam) { -- ythota death ener
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.25,
 
+    ---@param self ExperimentalPhasonLaserCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:404]: `Unit | Projectile | Prop | nil`

File: lua/defaultcollisionbeams.lua [hunk 14]
```diff
@@ -438,6 +462,9 @@ UltraChromaticBeamGeneratorCollisionBeam = Class(SCCollisionBeam) {
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.25,
 
+    ---@param self UltraChromaticBeamGeneratorCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:467]: `Unit | Projectile | Prop | nil`

File: lua/defaultcollisionbeams.lua [hunk 15]
```diff
@@ -504,6 +534,9 @@ TDFHiroCollisionBeam = Class(CollisionBeam) { -- used by UEF battlecruser
     SplatTexture = 'czar_mark01_albedo',
     ScorchSplatDropTime = 0.25,
 
+    ---@param self TDFHiroCollisionBeam
+    ---@param impactType string
+    ---@param targetEntity Unit
```
Comment (Garanas) [inline lua/defaultcollisionbeams.lua:539]: You get the picture, `targetEntity` in `OnImpact` should be `Unit | Projectile | Prop | nil`. Skipping the remainders (comment wise)

File: lua/defaultexplosions.lua [hunk 16]
```diff
@@ -464,7 +464,7 @@ end
 -- lot of overhead that is not necessary.
 ---@param unit Unit
 ---@param overKillRatio number
----@return unknown
+---@return Unit
```
Comment (Garanas) [inline lua/defaultexplosions.lua:467]: Returns an Entity, not a Unit

**Review by Garanas — CHANGES_REQUESTED** [review]
Great work, requires a few small changes.

**Comment by Garanas** [comment]
@Hdt80bro feel free to merge as you see fit

File: lua/formations.lua [hunk 17]
```diff
@@ -684,14 +693,18 @@ function PickBestTravelFormationIndex(typeName, distance)
...
 function PickBestFinalFormationIndex(typeName, distance)
     return -1;
 end
 
-
 -- ================ THE GUTS ====================
 -- ============ Formation Functions =============
 -- ==============================================
+---@param formationUnits Unit
+---@return true
```
Comment (Hdt80bro) [inline lua/formations.lua:707]: returns a `table`

File: lua/formations.lua [hunk 18]
```diff
@@ -1053,6 +1076,13 @@ function GetLandRowModifer(unitsList, categoryTable, currRowLen)
     end
 end
 
+---@param occupiedSpaces any
```
Comment (Hdt80bro) [inline lua/formations.lua:1079]: occupiedSpaces is a `boolean[][]`

File: lua/formations.lua [hunk 19]
```diff
@@ -1085,6 +1115,11 @@ function IsLandSpaceOccupied(occupiedSpaces, size, rowNum, whichCol, currRowLen,
     return false
 end
 
+---@param occupiedSpaces any
```
Comment (Hdt80bro) [inline lua/formations.lua:1118]: same

File: lua/formations.lua [hunk 20]
```diff
@@ -1426,11 +1475,9 @@ function GetChevronPosition(chevronPos, currCol, formationLen)
     return xPos, yPos
 end
 
-
-
-
-
 -- ========= UNIT SORTING ==========
+---@param unitsList Unit[]
+---@return Unit[]
```
Comment (Hdt80bro) [inline lua/formations.lua:1480]: Not actually a `Unit[]`; leave as `any`

File: lua/game.lua [hunk 21]
```diff
@@ -82,8 +87,10 @@ local bps = {
 -- Function for converting categories to string
 local ToString = import('/lua/sim/CategoryUtils.lua').ToString
 
--- Gets army index for specified army name
--- e.g. GetArmyIndex('ARMY_1') -> 1
+--- Gets army index for specified army name
+--- e.g. GetArmyIndex('ARMY_1') -> 1
+---@param armyName string
+---@return number|unknown
```
Comment (Hdt80bro) [inline lua/game.lua:93]: This function accepts an `Army` and returns `number`

File: lua/game.lua [hunk 22]
```diff
@@ -101,9 +108,11 @@ function GetArmyIndex(armyName)
...
 end
 
--- Adds restriction of units with specified Entity categories, e.g. 'categories.TECH2 * categories.AIR'
--- e.g. AddRestriction(categories.TECH2, 1) -> restricts all T2 units for army 1
--- e.g. AddRestriction(categories.TECH2) -> restricts all T2 units for all armies
+--- Adds restriction of units with specified Entity categories, e.g. 'categories.TECH2 * categories.AIR'
+--- e.g. AddRestriction(categories.TECH2, 1) -> restricts all T2 units for army 1
+--- e.g. AddRestriction(categories.TECH2) -> restricts all T2 units for all armies
+---@param cats EntityCategory
+---@param army number
```
Comment (Hdt80bro) [inline lua/game.lua:115]: army can be `Army` here

File: lua/game.lua [hunk 23]
```diff
@@ -118,9 +127,11 @@ function AddRestriction(cats, army)
...
 end
 
--- Removes restriction of units with specified Entity categories, e.g. 'categories.TECH1 * categories.UEF'
--- e.g. RemoveRestriction(categories.TECH2, 1) -> removes all T2 units restriction for army 1
--- e.g. RemoveRestriction(categories.TECH2) -> removes all T2 units restriction for all armies
+--- Removes restriction of units with specified Entity categories, e.g. 'categories.TECH1 * categories.UEF'
+--- e.g. RemoveRestriction(categories.TECH2, 1) -> removes all T2 units restriction for army 1
+--- e.g. RemoveRestriction(categories.TECH2) -> removes all T2 units restriction for all armies
+---@param cats EntityCategory
+---@param army number
```
Comment (Hdt80bro) [inline lua/game.lua:134]: army can be `Army` here

File: lua/game.lua [hunk 24]
```diff
@@ -162,17 +177,21 @@ function IsRestricted(unitId, army)
     return false
 end
 
--- Gets a table with ids of restricted units {Global = {}, PerArmy = {}}
+--- Gets a table with ids of restricted units {Global = {}, PerArmy = {}}
```
Comment (Hdt80bro) [inline lua/game.lua:180]: return `UnitRestrictions`; see below

File: lua/game.lua [hunk 25]
```diff
@@ -186,7 +205,7 @@ end
 -- Checks for valid unit blueprints (not projectiles/effects)
 local IsValidUnit = import('/lua/ui/lobby/UnitsAnalyzer.lua').IsValidUnit
 
--- Gets blueprints that can be upgraded, e.g. MEX, Shield, Radar structures
+--- Gets blueprints that can be upgraded, e.g. MEX, Shield, Radar structures
```
Comment (Hdt80bro) [inline lua/game.lua:208]: returns `UnitBlueprint[]`

File: lua/defaultantiprojectile.lua [hunk 26]
```diff
@@ -10,8 +10,30 @@
 local Entity = import('/lua/sim/Entity.lua').Entity
 local GetRandomFloat = import('/lua/utilities.lua').GetRandomFloat
 
+---@class FlareSpec
```
[the remark this answers is not in the document]
Comment (MrRowey) [reply lua/defaultantiprojectile.lua:13]: it this somthing that is TODO or somthing i need to add

[the remark this answers is not in the document]
Comment (Hdt80bro) [reply lua/defaultantiprojectile.lua:13]: If you'd like, you can tackle that or leave it as to-do

File: lua/game.lua [hunk 27]
```diff
@@ -90,7 +96,7 @@ local ToString = import('/lua/sim/CategoryUtils.lua').ToString
 --- Gets army index for specified army name
 --- e.g. GetArmyIndex('ARMY_1') -> 1
 ---@param armyName string
----@return number|unknown
+---@return Army|number
```
Comment (Hdt80bro) [inline lua/game.lua:99]: Sorry, I meant
```
---@param armyName Army
---@return number
```

File: lua/defaultunits.lua [hunk 28]
```diff
@@ -2547,7 +2550,7 @@ CommandUnit = Class(WalkingLandUnit) {
 
     ---@param self CommandUnit
     ---@param teleporter any
-    ---@param location number
+    ---@param location Vector
     ---@param orientation number
```
Comment (Hdt80bro) [inline lua/defaultunits.lua:2554]: orientation is `Quaternion`

File: lua/formations.lua [hunk 29]
```diff
@@ -908,10 +925,12 @@ function GuardFormation(formationUnits)
...
 end
 
-
-
-
 -- =========== LAND BLOCK BUILDING =================
+---@param unitsList Unit[]
+---@param formationBlock any
+---@param categoryTable EntityCategory[]
+---@param spacing any
```
Comment (Hdt80bro) [inline lua/formations.lua:932]: `spacing? number defaults to 1`

File: lua/formations.lua [hunk 30]
```diff
@@ -1124,11 +1162,11 @@ function GetColSpot(rowLen, col)
...
 end
 
-
-
-
-
 -- ============ AIR BLOCK BUILDING =============
+---@param unitsList Unit[]
+---@param airBlock any
+---@param spacing any
```
Comment (Hdt80bro) [inline lua/formations.lua:1168]: `spacing? number defaults to 1`

File: lua/formations.lua [hunk 31]
```diff
@@ -1234,6 +1272,9 @@ function BlockBuilderAir(unitsList, airBlock, spacing)
     return FormationPos
 end
 
+---@param unitsList Unit[]
+---@param spacing any
```
Comment (Hdt80bro) [inline lua/formations.lua:1276]: `spacing? number defaults to 1`

File: lua/game.lua [hunk 32]
```diff
@@ -24,18 +30,23 @@ VeteranDefault = {
...
+--- Modified to calculate the cost of an upgrade. The third argument is the economy section of
+--- the unit that is currently upgrading into the new unit. We subtract that cost from the cost
+--- of the unit that is being built
+---
+--- In order to keep backwards compatibility, there is a new option in the blueprint economy section.
+--- if DifferentialUpgradeCostCalculation is set to true, the base upgrade cost will be subtracted
+---@param builder Builder
+---@param targetData table
+---@param upgradeBaseData UnitBlueprintEconomy
+---@return number buildtimerate
```
Comment (Hdt80bro) [inline lua/game.lua:47]: let's call this `time`

File: lua/defaultunits.lua [hunk 33]
```diff
@@ -2507,7 +2510,7 @@ CommandUnit = Class(WalkingLandUnit) {
     end,
 
     ---@param self CommandUnit
-    ---@param bones string
+    ---@param bones Bone
```
Comment (Hdt80bro) [inline lua/defaultunits.lua:2513]: `Bone[]`

File: lua/formations.lua [hunk 34]
```diff
@@ -32,6 +32,9 @@ local FormationPos = {} -- list to be returned
 local FormationCache = {}
 local MaxCacheSize = 30
 
+---@param formationUnits Unit
```
Comment (Hdt80bro) [inline lua/formations.lua:35]: `Unit[]`

File: lua/formations.lua [hunk 35]
```diff
@@ -57,6 +60,9 @@ function GetCachedResults(formationUnits, formationType)
     return false
 end
 
+---@param results TLaserBotProjectile
+---@param formationUnits Unit
```
Comment (Hdt80bro) [inline lua/formations.lua:64]: `Unit[]`

File: lua/formations.lua [hunk 36]
```diff
@@ -684,14 +693,18 @@ function PickBestTravelFormationIndex(typeName, distance)
...
+---@return number
 function PickBestFinalFormationIndex(typeName, distance)
     return -1;
 end
 
-
 -- ================ THE GUTS ====================
 -- ============ Formation Functions =============
 -- ==============================================
+---@param formationUnits Unit
```
Comment (Hdt80bro) [inline lua/formations.lua:706]: `Unit[]`

File: lua/formations.lua [hunk 37]
```diff
@@ -748,6 +761,8 @@ function AttackFormation(formationUnits)
     return FormationPos
 end
 
+---@param formationUnits Unit
```
Comment (Hdt80bro) [inline lua/formations.lua:764]: `Unit[]`

File: lua/formations.lua [hunk 38]
```diff
@@ -822,6 +837,8 @@ function GrowthFormation(formationUnits)
...
```
Comment (Hdt80bro) [inline lua/formations.lua:840]: `Unit[]`

File: lua/formations.lua [hunk 39]
```diff
@@ -908,10 +925,12 @@ function GuardFormation(formationUnits)
     return FormationPos
 end
 
-
-
-
 -- =========== LAND BLOCK BUILDING =================
+---@param unitsList Unit[]
```
Comment (Hdt80bro) [inline lua/formations.lua:929]: This (and anything called `unitsList`) is the `table` we don't have a type for

File: lua/formations.lua [hunk 40]
```diff
@@ -1035,6 +1054,10 @@ function BlockBuilderLand(unitsList, formationBlock, categoryTable, spacing)
     return FormationPos
 end
 
+---@param unitsList Unit[]
```
Comment (Hdt80bro) [inline lua/formations.lua:1057]: `table` as well

File: lua/formations.lua [hunk 41]
```diff
@@ -1124,11 +1162,11 @@ function GetColSpot(rowLen, col)
     end
 end
 
-
-
-
-
 -- ============ AIR BLOCK BUILDING =============
+---@param unitsList Unit[]
```
Comment (Hdt80bro) [inline lua/formations.lua:1166]: `table`

File: lua/formations.lua [hunk 42]
```diff
@@ -1234,6 +1272,9 @@ function BlockBuilderAir(unitsList, airBlock, spacing)
     return FormationPos
 end
 
+---@param unitsList Unit[]
```
Comment (Hdt80bro) [inline lua/formations.lua:1275]: `table`

File: lua/formations.lua [hunk 43]
```diff
@@ -1338,6 +1379,9 @@ function BlockBuilderAirT3Bombers(unitsList, spacing)
...
```
Comment (Hdt80bro) [inline lua/formations.lua:1382]: `table`

File: lua/formations.lua [hunk 44]
```diff
@@ -1426,11 +1475,9 @@ function GetChevronPosition(chevronPos, currCol, formationLen)
     return xPos, yPos
 end
 
-
-
-
-
 -- ========= UNIT SORTING ==========
+---@param unitsList Unit[]
+---@return any
```
Comment (Hdt80bro) [inline lua/formations.lua:1480]: `table`

File: lua/formations.lua [hunk 45]
```diff
@@ -1508,6 +1555,8 @@ function CalculateSizes(unitsList)
     return unitsList
 end
 
+---@param formationUnits Unit
```
Comment (Hdt80bro) [inline lua/formations.lua:1558]: `Unit[]`

**Pull request merged**